distributed/shuffle/_scheduler_plugin.py

```diff
@@ -347,17 +389,28 @@ def transition(
         stimulus_id: str,
         **kwargs: Any,
     ) -> None:
+        """Clean up scheduler and worker state once a shuffle becomes inactive."""
         if finish not in ("released", "forgotten"):
             return
         if not key.startswith("shuffle-barrier-"):
             return
         shuffle_id = id_from_key(key)
         try:
-            shuffle = self.states[shuffle_id]
+            shuffle = self.active_shuffles[shuffle_id]
         except KeyError:
-            return
-        self._fail_on_workers(shuffle, message=f"{shuffle} forgotten")
-        self._clean_on_scheduler(shuffle_id)
+            pass
+        else:
+            self._fail_on_workers(shuffle, message=f"{shuffle} forgotten")
+            self._clean_on_scheduler(shuffle_id, stimulus_id=stimulus_id)
+
+        if finish == "forgotten":
+            shuffles = self._shuffles.pop(shuffle_id)
```

Comment:
IIUC this entire logic is just there to clean up. Behavior would not be impacted if we didn't do any of this, correct?

Reply:
Yes, this is state cleanup on the scheduler plugin.